pyscriptjs/src/plugins/pyterminal.ts

```diff
@@ -16,33 +29,31 @@ export class PyTerminalPlugin extends Plugin {
         this.app = app;
     }
 
-    configure(config: AppConfig & { terminal?: boolean | 'auto' }) {
+    configure(config: AppConfigStyle) {
         // validate the terminal config and handle default values
-        const t = config.terminal;
-        if (t !== undefined && t !== true && t !== false && t !== 'auto') {
-            const got = JSON.stringify(t);
-            throw new UserError(
-                ErrorCode.BAD_CONFIG,
-                'Invalid value for config.terminal: the only accepted' +
-                    `values are true, false and "auto", got "${got}".`,
-            );
-        }
+        const { terminal: t, docked: d } = config;
```

Comment:
ok, TIL a nice trick to read fields of a JS object :)

Reply:
> why are you using an anonymous function instead of using the function statement?

the correct terminology is *arrow function* because `function () {}` or `Function('')` are also anonymous (actually explicitly anonymous in the `Function` case).

Arrow functions have the following features:

  * these are not constructable so these are more explicit in their intent (helpers, utilities, not classes or functions with prototype): `new (() => {})` throws an error and no `prototype` exists at all
  * these do not pass through the context and arguments creation dance: both `this` and `arguments` don't exist, so these are considered lightweight functions
  * because of the previous point, if used within methods for any purpose, there's never the need to `.bind(this)`
  * minifiers produce smaller layout (as plain text)
  * last, but not least, these are shorter/faster to type, similarly like lambdas in python are used, whenever convenient

Function declarations are also hoisted, which is a whole shenanigan a part, while arrows are always assigned, example:

```js
// this works fine in JS
wut();

function wut() {
  console.log('wait ...');
}
```

```js
// this would throw
wut();

const wut = () => {
  console.log('wait ...');
};
```

So, for the same reason nobody uses `var` anymore, using functions when neither the `this` context nor `arguments` is needed, is consider legacy/anti-pattern and I hope the reasons I gave you make sense.

I've seen another file using *arrows* instead o function, but I would've asked soon otherwise why `function` declarations are used everywhere else instead ... it's not an issue in general, but there are extra footguns, variables shadowed, hoisting, unnecessary operations behind the scene, that if can be avoided, why not?

That's my thinking or feeling around this topic.